README.md

```diff
@@ -973,9 +974,9 @@ _concat([1, 2], 'buzz');
 
 The type of `_concat` is misleading: it suggests that it can operate on any
 two values of *any* one type. In fact there's an implicit constraint, since
-the type must support concatenation (in [mathematical][10] terms, the type
-must have a [semigroup][11]). The run-time type errors that result when this
-constraint is violated are not particularly descriptive:
+the type must support concatenation (in [mathematical][semigroup] terms, the
+type must have a [semigroup][Semigroup]). The run-time type errors that result
+when this constraint is violated are not particularly descriptive:
```

Comment:
Before I merge this pull request I'll confirm that link identifiers are case sensitive.